### ISSUE TITLE
SampledProperty.interpolationAlgorithm should be labelled read-only

### ISSUE THREAD

**Issue by OmarShehata** [CesiumGS/cesium#7597]
If you read the docs: https://cesiumjs.org/Cesium/Build/Documentation/SampledProperty.html?classFilter=Sampled#interpolationAlgorithm you might be encouraged to go around setting this property in your applications, trusting that everything will work as intended.

The docs give you false hope however, since this property does not have a setter:

https://github.com/AnalyticalGraphicsInc/cesium/blob/1.54/Source/DataSources/SampledProperty.js#L287

You'll get:

```
TypeError: setting getter-only property "interpolationAlgorithm"
```

Also, [EasingFunction](
https://cesiumjs.org/Cesium/Build/Documentation/EasingFunction.html) links to TweenCollection which doesn't seem to exist:

https://cesiumjs.org/Cesium/Build/Documentation/TweenCollection.html

I ran into all this trying to have a `SampledProperty` do an easing instead of just linearly interpolating. Is there no easy way to do this?

**Comment by hpinkos** [comment]
@OmarShehata `TweenCollection` is a private class, so `EasingFunction` should probably either be private as well, or shouldn't link to `TweenCollection`.

Currently `SampledProperty` supports Linear, Lagrange, or Hermite interpolation.  See https://cesiumjs.org/Cesium/Build/Apps/Sandcastle/?src=Interpolation.html

**Comment by OmarShehata** [comment]
I feel like we should expose `TweenCollection` as public. Given that we ship our library with TweenJS, it'd be nice to expose it. Otherwise, a user would have to include their copy of it to do some of this common operations. 

This came up in this thread on tweening some entity properties on user interaction:

https://groups.google.com/d/msg/cesium-dev/shXInRucRtY/9nne8r4PBwAJ

In this case, it's things like having a billboard fade in/out on mouse over for example, which I think is conceptually different from time-dynamic data that should be tied to the timeline.

**Comment by mramato** [comment]
@OmarShehata Users might need additional API to implement some use case, but simply exposing TweenCollection and calling it a day would be a leaky abstraction.

That being said, this issue is about fixing the documentation, so let's not get sidetracked with other things (especially if this is going to be labeled `good first issue`.  Please write up a new issue if you feel there needs to be a feature request to do something.

**Comment by hpinkos** [comment]
Fixed in #8821

**Comment by cesium-concierge** [comment]
Congratulations on closing the issue! I found these Cesium forum links in the comments above:

https://groups.google.com/d/msg/cesium-dev/shXInRucRtY/9nne8r4PBwAJ

If this issue affects any of these threads, please post a comment like the following:

> The issue at https://github.com/CesiumGS/cesium/issues/7597 has just been closed and may resolve your issue. Look for the change in the next stable release of Cesium or get it now in the master branch on GitHub https://github.com/AnalyticalGraphicsInc/cesium.
